src-java/kilda-model/src/main/java/org/openkilda/model/bitops/BitField.java

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.model.bitops;
+
+import org.openkilda.model.Cookie;
+
+import lombok.Getter;
+
+@Getter
+public class BitField {
+    private final long mask;
+    private final int offset;
+
+    public BitField(long mask) {
```

Comment:
nit: Can't this be just as following?
```
if (mask == 0) throw new IllegalArgumentException("Bit field mask must not be 0");
long theBit = Long.lowestOneBit(mask);
if (theBit != mask) throw new IllegalArgumentException("Illegal bit field mask");
offset = Long.numberOfTrailingZeros(theBit) + 1;
```

Reply:
`Long.lowestOneBit(mask)` return one lowest bit, so this approach fails for any bit field(mask) longer than 1 bit.

Reply:
Got it.